cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -461,5 +487,26 @@ mod test {
             // Empty entries should be removed
             prop_assert!(reg.entry_for_hash(&hash).is_none());
         }
+
+        #[test]
+        fn garbage_collect_envs(mut reg: EnvRegistry) {
+            prop_assume!(!reg.entries.is_empty());
+            prop_assert!(reg.entries.iter().all(|entry| !entry.exists()));
+
+            // Add a single environment with a real path on disk.
+            let (flox, _temp_dir) = flox_instance();
+            let reg_path = env_registry_path(&flox);
+            let env = new_path_environment(&flox, "version = 1");
+            let env_hash = path_hash(&env.path);
+            reg.register_env(&env.path, &env_hash, &EnvironmentPointer::Path(env.pointer)).unwrap();
+            write_environment_registry(&reg, &reg_path, acquire_env_registry_lock(&reg_path).unwrap()).unwrap();
+            prop_assert!(reg.entries.len() > 1);
```

Comment:
question blocking: is the same environment getting double registered, once by `new_path_environment` and once by `register_env`? Dunno if I'm missing something but it's confusing to read `Add a single environment with a real path on disk` and `prop_assert!(reg.entries.len() > 1);`

If the same env is getting double registered, I think it would be better to just use one env, `garbage_collect` and assert it didn't get removed, then remove the env from disk, `garbage_collect` again, and assert it's now removed

Reply:
Yeah, I'd forgotten that the helper behaves differently from `init` which doesn't call `open` and `ensure_registered`.

Reply:
Done (to squash)